### PR TITLE
Add shortNames for various CRDs

### LINKED ISSUE #600: [Feature Request] Add short names for CRDs
### Description of problem

The names of CRDs in cartographer are quite long and typing them out constantly can be tedious.

### Proposed solution

Add Short names to the CRDs of cartographer to allow for quick access to crds via easy to understand short names

### Example

Clustersupplychain -> csc
Clustertemplate -> ct
Clusterimagetemplate -> cit
Clusterconfigtemplate -> cct
Clustersourcetemplate -> cst
Clusterdeploymenttemplate -> cdt
Clusterruntemplate -> crt
clusterdelivery -> cd
runnable -> run
Workload -> wld
Delivery -> dlv

## Additional Context

The shorter names such as workload,runnable and deliverable are less important to have these but for the templates and supply chain it would enhance the UX greatly.

### PR DESCRIPTION
**Pull request by benjaminguttmann-avtq**
[PR requirements]: https://github.com/vmware-tanzu/cartographer/blob/main/CONTRIBUTING.md#commit-message-and-pr-guidelines

## Changes proposed by this PR

closes #600 

Added shortNames for the different CRDs according to the suggestion from the issue #600.

Not sure if those should be added in the documentation somewhere.

## Release Note

Adds shortNames for ClusterConfigTemplates, ClusterDeliveries, ClusterDeploymentTemplates, ClusterImageTemplates, ClusterRunTemplates, ClusterSourceTemplates, ClusterSupplyChain, ClusterTemplates, Deliverables, Runnables, and Workloads

[imperative mood]: https://chris.beams.io/posts/git-commit/#imperative

## PR Checklist

Note: Please do not remove items. Mark items as done `[x]` or use ~strikethrough~ if you believe they are not relevant

- [X] Linked to a relevant issue. Eg: `Fixes #123` or `Updates #123`
- [X] Removed non-atomic or `wip` commits
- [X] Filled in the [Release Note](#Release-Note) section above 
- ~~[ ] Modified the docs to match changes ~~

### CODE REVIEW

File: config/crd/bases/carto.run_deliverables.yaml [hunk 1]
```diff
@@ -28,6 +28,8 @@ spec:
     kind: Deliverable
     listKind: DeliverableList
     plural: deliverables
+    shortNames:
+    - cit
```
Comment (martyspiewak) [inline config/crd/bases/carto.run_deliverables.yaml]:
```suggestion
    - dlv
```

File: pkg/apis/v1alpha1/deliverable.go [hunk 2]
```diff
@@ -61,7 +61,7 @@ var ValidDeliverablePrefixes = []string{
 }
 
 // +kubebuilder:object:root=true
-// +kubebuilder:resource:categories=all
+// +kubebuilder:resource:path=deliverables, categories=all, shortName=cit
```
Comment (martyspiewak) [inline pkg/apis/v1alpha1/deliverable.go]:
```suggestion
// +kubebuilder:resource:path=deliverables,categories=all,shortName=dlv
```

File: pkg/apis/v1alpha1/runnable.go [hunk 3]
```diff
@@ -41,6 +41,7 @@ const (
 
 // +kubebuilder:object:root=true
 // +kubebuilder:subresource:status
+// +kubebuilder:resource:shortName=run
```
Comment (martyspiewak) [inline pkg/apis/v1alpha1/runnable.go]:
```suggestion
// +kubebuilder:resource:path=runnables,shortName=run
```

File: pkg/apis/v1alpha1/workload.go [hunk 4]
```diff
@@ -67,8 +67,8 @@ var ValidWorkloadPrefixes = []string{
 }
 
 // +kubebuilder:object:root=true
-// +kubebuilder:resource:categories="all"
 // +kubebuilder:subresource:status
+// +kubebuilder:resource:path=workloads, shortName="wld", categories="all"
```
Comment (martyspiewak) [inline pkg/apis/v1alpha1/workload.go]:
```suggestion
// +kubebuilder:resource:path=workloads,shortName=wld,categories=all
```

**Review by martyspiewak — CHANGES_REQUESTED** [review]
Overall this looks great- thanks for submitting!

I left a couple of comments where i think you had a typo (`cit` instead of `dlv`). I also suggested removing the spaces in the kubebuilder annotations there as we don't generally have spaces. Would you mind removing the spaces from all the kubebuilder annotations you added (i didnt get a chance to leave a comment on every file that has spaces)?

**Comment by benjaminguttmann-avtq** [comment]
@martyspiewak thx for the review and the feedback, adjusted accordingly.

**Review by martyspiewak — APPROVED** [review]
Looks great! Thanks so much for submitting!

**Pull request merged**